ludwig/features/image_feature.py

```diff
@@ -50,6 +51,62 @@ def get_feature_meta(column, preprocessing_parameters):
             'preprocessing': preprocessing_parameters
         }
 
+    @staticmethod
+    def _read_image_and_resize(filepath, im_width, im_height,
+                               should_resize, num_channels,
+                               resize_method, user_specified_num_channels):
+        """
+        :param filepath: path to the image
+        :param im_width: expected width of the image
+        :param im_height: expected height of the image
+        :param should_resize: Should the image be resized?
+        :param resize_method: type of resizing method
+        :param num_channels: expected number of channels in the first image
+        :param user_specified_num_channels: did the user specify num channels?
```

Comment:
can we get rid of this variable and have num_channels with None initializer and then check 
if num_channels is not None instead?

Reply:
Yeah, I tried that actually. It failed because we're updating the metadata which is in turn updating the preprocessing parameters. We will lose track of the fact that the user didn't actually mention a value.